src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -55,14 +55,16 @@ public function getArguments(Request $request, $controller)
 
                 $resolved = $resolver->resolve($request, $metadata);
 
-                if (!$resolved instanceof \Generator) {
-                    throw new \InvalidArgumentException(sprintf('%s::resolve() must yield at least one value.', \get_class($resolver)));
```

Comment:
checking for Generator does not achieve the intended goal: having at least one yielded value.
I can do `return; yield $value;` which would return a generator but not yield anything.

Reply:
I don't see a return type on the interface, so this would allow non-iterable return types as well. Perhaps the interface should receive a return type to enforce this (which is `\Generator` at the moment, `iterable` after this PR) in 5.0? Could do with a typecheck on `$resolved`

Another solution would be to allow a `return $value` in the resolvers as generators are pretty much only required for an array structure being inserted as multiple arguments.

Reply:
There are several different topics:
1. adding real return types: I would also prefer this. But not sure if we will do that in SF because it's a hard break for all implementations (even with return type variance in PHP 7.4). So that is a different topic.
2. Validating returned type: Could make sense. But normally we do not do that for every method as we would just need to check types everywhere. So if you return something non-iterable and violate the interface, the code will just break. No real need to validate that.
3. Changing the return type: I also think instead of forcing a iterable/Generator everywhere, it should allow only returning a single value (which is the common need). For variadic arguments, there could just be an explicit `VariadicArgumentValue` class that you can return to resolve to several arguments. I think that's more clear and clean. Currently you can yield several values for non-variadic arguments which makes the ArgumentResolver behave very strange. But that's also a different topic. If you argree with something like this, feel free to create an issue/PR to keep track of that.